app/javascript/App/bundles/NewsfeedPage/NewsList.tsx

```diff
@@ -0,0 +1,187 @@
+declare var window: {
+  isMobile?: boolean,
+  isTablet?: boolean,
+};
+
+import * as React from 'react'
+import InfiniteScroll from 'react-infinite-scroll-component'
+import _ from 'lodash'
+import NewsListItem from './NewsListItem'
+import LoadingIndicator from '../../components/LoadingIndicator'
+import Tips from './Tips'
+import scrollHelper from './../../scrollHelper'
+
+import { NewsItem } from './types';
+
+interface Props {
+  newsItems: Array<NewsItem>,
+  isLoading: () => boolean,
+  activeFilters: any,
+  sortedNewsItems: any,
+  initialRenderTips: boolean,
+  fetchMoreNewsFeed: () => void,
+  toggleNewsfeedTips: () => void,
+};
+
+interface State {
+  initialRender: boolean,
+  initialRenderTips: boolean,
+};
+
+class NewsList extends React.Component<Props, State> {
+  state = { initialRender: true, initialRenderTips: false }
+
+  componentDidMount() {
+    // set max height to enable scroll in ff
+    // scrollHelper() // TODO: check, it not working
```

Comment:
Can't say now. It was something like `style of undefined` in `scrollHelper` on line 26. I suppose this happens because this code is not yet refactored. I started to work on it today